fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java

```diff
@@ -85,46 +75,95 @@ public void doExecute() throws Exception {
      * 3. insert col stats and partition stats
      */
     protected void doSample() throws Exception {
-        Pair<List<Long>, Long> pair = calcActualSampleTablets();
+        LOG.info(String.format("Will do sample collection for column %s", col.getName()));
+        Pair<List<Long>, Long> pair = calcActualSampleTablets(isPartitionColumn());
+        LOG.info(String.format("Number of tablets selected %d, rows in tablets %d", pair.first.size(), pair.second));
         List<Long> tabletIds = pair.first;
         double scaleFactor = (double) tbl.getRowCount() / (double) pair.second;
         // might happen if row count in fe metadata hasn't been updated yet
         if (Double.isInfinite(scaleFactor) || Double.isNaN(scaleFactor)) {
+            LOG.warn("Scale factor is infinite or Nan, will set scale factor to 1.");
```

Comment:
Under what circumstances does this problem occur?

Reply:
It might happen if row count in fe metadata hasn't been updated yet